.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+name: CI
+on: [push, pull_request, workflow_call, workflow_dispatch]
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set CACHE_DATE
+      run: eval echo 'CACHE_DATE="$(date --utc +%Y%m%dT%H%M%SZ)"' >> $GITHUB_ENV
+    - run: ./devel/pull
+    - run: ./devel/build
+    - if: ${{ github.event_name == 'push' }}
+      name: Docker login
+      run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+    - if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref_name == github.event.repository.default_branch }}
+      run: |
+        ./devel/tag build-$CACHE_DATE
+        ./devel/push latest build-$CACHE_DATE
+    - if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref_name != github.event.repository.default_branch }}
+      run: |
+        tag=branch-${GITHUB_REF_NAME//[^A-Za-z0-9._-]/-}
```

Comment:
Would be nice to preserve the comment from `push-from-travis` which explains the substitution here.

Reply:
Done.